src/ray/gcs/pb_util.h

```diff
@@ -337,6 +339,43 @@ inline void FillTaskStatusUpdateTime(const ray::rpc::TaskStatus &task_status,
   }
 }
 
+/// Generate a placement constraint for placement group.
+///
+/// \param pg_id The ID of placement group.
+/// \param strategy The placement strategy of placement group.
+/// \return The placement constraint for placement group if it's not a strict
+///   strategy, else absl::nullopt.
+inline absl::optional<rpc::autoscaler::PlacementConstraint>
+GenPlacementConstraintForPlacementGroup(const std::string &pg_id,
+                                        rpc::PlacementStrategy strategy) {
+  // Defaults to be overwritten.
+  rpc::autoscaler::PlacementConstraint::Type type =
+      rpc::autoscaler::PlacementConstraint::ANTI_AFFINITY;
+  switch (strategy) {
+  case rpc::PlacementStrategy::STRICT_SPREAD: {
+    type = rpc::autoscaler::PlacementConstraint::ANTI_AFFINITY;
+    break;
+  }
+  case rpc::PlacementStrategy::STRICT_PACK: {
+    type = rpc::autoscaler::PlacementConstraint::AFFINITY;
+    break;
+  }
+  case rpc::PlacementStrategy::SPREAD:
+  case rpc::PlacementStrategy::PACK: {
+    return absl::nullopt;
+  }
+  default: {
+    RAY_LOG(ERROR) << "Encountered unexpected strategy type: " << strategy;
+  }
+  }
+  auto pg_constraint = rpc::autoscaler::PlacementConstraint();
+  pg_constraint.set_type(type);
+  pg_constraint.set_key(kPlacementGroupConstraintKey);
```

Comment:
we need to set pg_id as key since we doesn't allow multiple values for the same label name.